code/controllers/subsystem/research.dm

```diff
@@ -86,17 +86,46 @@ SUBSYSTEM_DEF(research)
 	return SS_INIT_SUCCESS
 
 /datum/controller/subsystem/research/fire()
+	var/can_synchronize=FALSE
+
+	//since the game will ever have only 1 master server we just return if that is offline
+	for(var/obj/machinery/rnd/server/master/master_server in master_servers)
+		can_synchronize=TRUE
+		if(!master_server.working)
+			///we still want to initialize the last time this method was called to prevent point gaines during the time the server was off
+			last_income = world.time
+			return
```

Comment:
this isn't a bugfix, it's a balance change.
R&D is supposed to work as long as there's one server available. The master server is there for traitor objectives

Reply:
Your right the bug fix isn't about the server its about component printer and it isn't related .

 I just added master server reliance as a feature and as I said in my PR servers depend on master on/off state only if the master server exists.

If master is destroyed then we go back to the old behavior of requiring only one server for functioning and points still continue to be generated at half the rate. plzz look at video 3

Reply:
how so? If the master server isn't working, it early returns.
If there is no master server, it works? This is inconsistent

Reply:
Alright I'll remove the dependence